service/controller/v25/resource/routetable/resource.go

```diff
@@ -0,0 +1,132 @@
+package routetable
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+
+	"github.com/giantswarm/aws-operator/service/controller/v25/controllercontext"
+)
+
+const (
+	Name = "routetablev25"
+)
+
+type Config struct {
+	Logger micrologger.Logger
+
+	Names []string
+}
+
+type Resource struct {
+	logger micrologger.Logger
+
+	// mappings is a mapping of route table names and IDs, where the key is the
+	// name and the value is the ID.
+	mappings map[string]string
+	mutex    sync.Mutex
+
+	names []string
+}
+
+func New(config Config) (*Resource, error) {
+	if config.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Logger must not be empty", config)
+	}
+
+	if config.Names == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Names must not be empty", config)
+	}
+
+	r := &Resource{
+		logger: config.Logger,
+
+		mappings: map[string]string{},
+		mutex:    sync.Mutex{},
+
+		names: config.Names,
+	}
+
+	return r, nil
+}
+
+func (r *Resource) Name() string {
+	return Name
+}
+
+func (r *Resource) addRouteTableMappingsToContext(ctx context.Context) error {
+	r.mutex.Lock()
+	defer r.mutex.Unlock()
```

Comment:
I'm fine with having a mutex here, but curious how this can get called concurrently?

Reply:
Not really right now. I just want to guard the caching a bit. Just in case.